heartbeat/monitors/active/http/check.go

```diff
@@ -132,14 +135,85 @@ func checkHeaders(headers map[string]string) respValidator {
 	}
 }
 
-func checkBody(matcher []match.Matcher) bodyValidator {
+func parseBody(b interface{}) (positiveMatch, negativeMatch []match.Matcher) {
+	// run through this code block if there is no positive or negative keyword in response body
+	// in this case, there's only plain body
+	if p, ok := b.([]interface{}); ok {
+		for _, pp := range p {
+			if pat, ok := pp.(string); ok {
+				return append(positiveMatch, match.MustCompile(pat)), negativeMatch
+			}
+		}
+	}
+
+	// run through this part if there exists positive/negative keyword in response body
+	// in this case, there will be 3 possibilities: positive + negative / positive / negative
+	if m, ok := b.(map[string]interface{}); ok {
+		for checkType, v := range m {
+			if params, ok := v.([]interface{}); ok {
+				for _, param := range params {
+					if pat, ok := param.(string); ok {
+						if checkType == "positive" {
+							positiveMatch = append(positiveMatch, match.MustCompile(pat))
+						} else if checkType == "negative" {
+							negativeMatch = append(negativeMatch, match.MustCompile(pat))
+						} else {
+							return positiveMatch, negativeMatch
```

Comment:
This seems more like an error condition, and one that could be caught earlier (in the `for checkType loop`). In the case of error conditions (a type like `ppositive`) we should try and help the user out and report an error like "invalid check type 'ppositive', expected 'positive' or 'negative'.